src/lib/Strategy/FocusMode/ContentStructureRedirectStrategy.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\AdminUi\Strategy\FocusMode;
+
+use Ibexa\Contracts\AdminUi\FocusMode\RedirectStrategyInterface;
+use Ibexa\Contracts\Core\Repository\LocationService;
+use Ibexa\Contracts\Core\SiteAccess\ConfigResolverInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+final class ContentStructureRedirectStrategy implements RedirectStrategyInterface
+{
+    private ConfigResolverInterface $configResolver;
+
+    private LocationService $locationService;
+
+    private RouterInterface $router;
+
+    public function __construct(
+        ConfigResolverInterface $configResolver,
+        LocationService $locationService,
+        RouterInterface $router
+    ) {
+        $this->configResolver = $configResolver;
+        $this->locationService = $locationService;
+        $this->router = $router;
+    }
+
+    /**
+     * Contains paths that are not available in "Focus mode" so shouldn't be redirected to.
+     *
+     * @param array<string, string> $routeData
+     */
+    public function supports(array $routeData): bool
+    {
+        ['_route' => $route] = $routeData;
+
+        return in_array(
+            $route,
+            [
+                'ibexa.section.list',
+                'ibexa.content_type_group.list',
+                'ibexa.object_state.groups.list',
+                'ibexa.content_type_group.view',
```

Comment:
Shouldn't this list be configurable?

Reply:
To me it shouldn't - it seems like an extension point nobody would really use. The thing is, we are lacking a proper list of columns/paths unavailable in focus mode which can be referenced here. That's why I decided to hardcode them and give way to circumvent the default behavior by registering another strategy at the same time. To me, Open-Closed principle holds it ground in this case. 😉